src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -262,20 +262,23 @@ public static bool IsArrayTypeResponseOperation(Operation op, ServiceDefinition
             if (op.Responses["200"]?.Schema?.Reference?.Equals(string.Empty) == false)
             {
                 var modelLink = op.Responses["200"].Schema.Reference;
+
+                var def = entity.Definitions.GetValueOrNull(modelLink.StripDefinitionPath());
+
                 // if the object has more than 2 properties, we can assume its a composite object
                 // that does not represent a collection of some type
-                if ((entity.Definitions[modelLink.StripDefinitionPath()].Properties?.Values?.Count ?? 2) >= 2)
+                if ((def?.Properties?.Values?.Count ?? 2) >= 2)
                 {
                     return false;
                 }
 
                 // if the object is an allof on some other object, let's consider it to be a composite object
-                if (entity.Definitions[modelLink.StripDefinitionPath()].AllOf != null)
+                if (def.AllOf != null)
```

Comment:
If `def` were `null`, the previous check (`(def?.Properties?.Values?.Count ?? 2) >= 2`) is `true` and exits the function early.